src/plugins/dashboard/public/dashboard_app/top_nav/dashboard_top_nav.tsx

```diff
@@ -116,6 +123,19 @@ export function DashboardTopNav({ embedSettings, redirectTo }: DashboardTopNavPr
     if (!embedSettings) setChromeVisibility(viewMode !== ViewMode.PRINT);
   }, [embedSettings, setChromeVisibility, viewMode]);
 
+  /**
+   * Keep track of the height of the top nav bar as it changes so that the padding at the top of the
+   * dashboard viewport can be adjusted dynamically as it changes
+   */
+  const resizeRef = useRef<HTMLDivElement>(null);
+  const dimensions = useResizeObserver(resizeRef.current);
+  useEffect(() => {
+    if (dimensions.height !== currentHeight) {
```

Comment:
What is the purpose of keeping this state both in the `top_nav` and also in the `dashboard_app`? I see that you're using it to ensure that the callback which sets the `dashboard_app` version only gets updated when the height has actually changed, but wouldn't that already be accomplished via the resize observer? Or is the resize observer firing all the time and causing unnecessary re-renders?

If the resize observer works like I think, it might be cleaner to just call `onHeightChange` every time without diffing or keeping another piece of state.

Reply:
That's a fair point! I added that because the resize observer fires for **all** resize events, including width; however, I wanted to update the top padding only when the **height** of the `div` changed. It was just a tiny efficiency improvement - tested though, and I don't think it significantly impacts the number of renders for the `dashboard_app` or anything :eyes:

I think you're right that duplicating the state like this isn't super clean - I can remove it in favour of just calling `onHeightChange` for every resize :+1: